sdk/client/client.ts

```diff
@@ -124,39 +117,98 @@ export class Client {
   basePath: string
   // The available API endpoints
   api: Api
+
   // The path to the secrets file which may be used to store credentials
-  secretsFilePath: string
+  secretsFilePath: string = process.cwd() + '/secrets.json'
+  secretsFileContent: { api: {} }
+
   // The credential access details
   private tokenUrlDetails: [Source, string]
   private usernameDetails: [Source, string]
   private passwordDetails: [Source, string]
   private clientIdDetails: [Source, string]
   private clientSecretDetails: [Source, string]
+
   // The refresh limit in seconds before token expiry to trigger a refresh
-  refreshLimit: number = refreshLimit
+  refreshLimit: number = 3580
```

Comment:
Should we also allow providing a custom refreshLimit with this set as a sensible default?